prometheus/promhttp/http.go

```diff
@@ -54,6 +55,8 @@ const (
 	processStartTimeHeader = "Process-Start-Time-Unix"
 )
 
+var defaultEncodingOffers = []string{"gzip", "zstd"}
```

Comment:
I propose we use enum here e.g `Encoding`, also add `identity` as a clear options caller can use for no compression as per RFC 9110